src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs

```diff
@@ -469,8 +482,19 @@ private bool LaunchHost(TestProcessStartInfo testHostStartInfo, CancellationToke
             _processHelper.SetExitCallback(processId, ExitCallBack);
         }
 
+        if (_testHostProcess is null)
+        {
+            return false;
+        }
+
+        _testHostProcess.PriorityClass =
+            _environmentVariableHelper.GetEnvironmentVariable("VSTEST_HOST_PRIORITY") is { } processPriorityString
```

Comment:
Is there a better syntax than `{ }` to get non-null value? e.g. `is string` or `is not null`?

Maybe it would be easier to understand as

Or _environmentVariableHelper.GetEnvironmentVariableAsEnum<PriorityClass?>("VSTEST_HOST_PRIORITY", defaultValue: null); 

Because that would be useful utility for other places where we want to get value from the env variable and provide default value as well, no matter if null, or actual fallback value like PriorityClass.Normal (which we don't want to do here to avoid touching the priority unless user specifically asked us to set it).

The env variable should also allow

Reply:
Seems like even more generic GetEnvironmentVariable<T> would be also useful, but is there a universal interface we can use for that like IParseable, or some converter? Are type constraints like: where T is int or is double or is string or is enum possible nowadays?

Reply:
> Is there a better syntax than `{ }` to get non-null value? e.g. `is string` or `is not null`?

Yes `is string processPriorityString`  would work.

> Seems like even more generic GetEnvironmentVariable would be also useful, but is there a universal interface we can use for that like IParseable, or some converter? Are type constraints like: where T is int or is double or is string or is enum possible nowadays?

I will make some test but I think if `T : IConvertible` should be enough.